src/core/processing/processingalgorithmparametersmodel.cpp

```diff
@@ -0,0 +1,254 @@
+/***************************************************************************
+  processingalgorithmparametersmodel.h - ProcessingAlgorithmParametersModel
+
+ ---------------------
+ begin                : 21.06.2024
+ copyright            : (C) 2024 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "processingalgorithmparametersmodel.h"
+
+#include <qgsapplication.h>
+#include <qgsprocessingalgorithm.h>
+#include <qgsprocessingparameters.h>
+#include <qgsprocessingregistry.h>
+
+
+ProcessingAlgorithmParametersModel::ProcessingAlgorithmParametersModel( QObject *parent )
+  : QSortFilterProxyModel( parent )
+  , mModel( new ProcessingAlgorithmParametersModelBase( parent ) )
+{
+  setSourceModel( mModel );
+  connect( mModel, &ProcessingAlgorithmParametersModelBase::algorithmIdChanged, this, &ProcessingAlgorithmParametersModel::algorithmIdChanged );
+  connect( mModel, &ProcessingAlgorithmParametersModelBase::parametersChanged, this, &ProcessingAlgorithmParametersModel::parametersChanged );
+}
+
+void ProcessingAlgorithmParametersModel::setFilters( ProcessingAlgorithmParametersModel::Filters filters )
+{
+  if ( mFilters == filters )
+  {
+    return;
+  }
+
+  mFilters = filters;
+  emit filtersChanged();
+
+  invalidateFilter();
+}
+
+QString ProcessingAlgorithmParametersModel::algorithmId() const
+{
+  return mModel->algorithmId();
+}
+
+void ProcessingAlgorithmParametersModel::setAlgorithmId( const QString &id )
+{
+  mModel->setAlgorithmId( id );
+}
+
+bool ProcessingAlgorithmParametersModel::isValid() const
+{
+  return mModel->isValid();
+}
+
+bool ProcessingAlgorithmParametersModel::hasAdvancedParameters() const
+{
+  return mModel->hasAdvancedParameters();
+}
+
+QString ProcessingAlgorithmParametersModel::algorithmDisplayName() const
+{
+  return mModel->algorithmDisplayName();
+}
+
+QString ProcessingAlgorithmParametersModel::algorithmShortHelp() const
+{
+  return mModel->algorithmShortHelp();
+}
+
+QVariantMap ProcessingAlgorithmParametersModel::parameters()
+{
+  return mModel->parameters();
+}
+
+void ProcessingAlgorithmParametersModel::setParameters( const QVariantMap &parameters )
+{
+  mModel->setParameters( parameters );
```

Comment:
emit changed

Reply:
That's taken care through theses connected signals:

```
  connect( mModel, &ProcessingAlgorithmParametersModelBase::algorithmIdChanged, this, &ProcessingAlgorithmParametersModel::algorithmIdChanged );
  connect( mModel, &ProcessingAlgorithmParametersModelBase::parametersChanged, this, &ProcessingAlgorithmParametersModel::parametersChanged );
```